### PR TITLE
feat(addon): #732 Create experiment for DeleteMenu

### LINKED ISSUE #732: Create Experiment for Delete/Block Menu Item Order
This experiment will simply reverse the order of the 'X' menu items (block/delete) in 10% of the population.

1.  instrument addon so that on start-up, it will determine if it is part of the experiment (10%) or control (90%) and persist the result (based on hash of client_id?)
2.  assign an experiment_id to uniquely identify the test (hardcoded)
3.  if the experiment is enabled, reverse the 'X' menu items
4.  if the experiment is enabled, when sending event pings, make sure to include the experiment_id field
5.  modify data pipeline / infernyx job / database schema to process and persist experiment_id
6.  add a comparative re:dash graph to show any differences between the experiment and it's control

### PR DESCRIPTION
**Pull request by ncloudioj**
Fixes #732 

This change is [Reviewable](https://reviewable.io/reviews/mozilla/activity-stream/777)

### CODE REVIEW

File: content-src/components/DeleteMenu/DeleteMenu.js [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
+      let payload = {
         event,
         page: this.props.page,
         source: this.props.source,
         action_position: this.props.index
-      }));
+      };
+
+      if (this.props.experimentData.reverseMenuOptions) {
+        payload.experiment_id = this.props.experimentData.id;
```
Comment (k88hudson) [inline content-src/components/DeleteMenu/DeleteMenu.js]: This should always be included I think, not just for a truthy value

File: lib/ActivityStreams.js [hunk 2]
```diff
@@ -60,6 +61,17 @@ function ActivityStreams(options = {}) {
   this._memoized = this._get_memoized(this._memoizer);
 
   this._telemetrySender = new TelemetrySender();
+
+  this._experimentData = {};
+  let rng = seedrandom(options.clientID);
+  // Targeting 10% of users for this experiment
+  if (rng() <= 0.1) {
+    this._experimentData.id = "exp-001-delete-menu";
+    this._experimentData.reverseMenuOptions = true;
```
Comment (k88hudson) [inline lib/ActivityStreams.js]: I would recommend using string or integer based variant IDs such as `"CONTROL"` and `"REVERSE"`.  Also, you might want to call the experiment something neutral like `deleteMenuOrder`

File: content-src/components/DeleteMenu/DeleteMenu.js [hunk 3]
```diff
@@ -57,4 +65,6 @@ DeleteMenu.propTypes = {
   index: React.PropTypes.number
 };
 
-module.exports = connect(justDispatch)(DeleteMenu);
+module.exports = connect(state => {
+  return {experimentData: state.Experiments.data};
+})(DeleteMenu);
```
Comment (k88hudson) [inline content-src/components/DeleteMenu/DeleteMenu.js]: Could we have a test for this?

**Comment by k88hudson** [comment]
Ok, we are going to do a follow-up patch to refactor the randomizer stuff into its own module

File: lib/ActivityStreams.js [hunk 4]
```diff
@@ -60,6 +61,17 @@ function ActivityStreams(options = {}) {
   this._memoized = this._get_memoized(this._memoizer);
 
   this._telemetrySender = new TelemetrySender();
+
+  this._experimentData = {};
+  let rng = seedrandom(options.clientID);
+  // Targeting 10% of users for this experiment
```
Comment (yati-sagade) [inline lib/ActivityStreams.js]: Hi! Just a curious Joe here. Did this code run on every new tab? If so, is this not putting 10% of new tab opens (as opposed to 10% of FF eligible FF installs) in the variant?

Comment (yati-sagade) [reply lib/ActivityStreams.js]: Just saw the seedrandom(), so that should be ok :)

**Pull request merged**